docs/developer/decision-records/2022-02-03-integration-testing/README.md

```diff
@@ -0,0 +1,302 @@
+    implementation(project(":samples:04.0-file-transfer:provider"))
+```
+
+The JUnit integration test runs the Provider using the preexisting `EdcExtension`, and the Consumer using a newly developed extension:
+
+```java
+    // EDC Consumer runtime
+    @RegisterExtension
+    @Order(1)
+    static GradleModuleRuntimeExtension otherConnector = new GradleModuleRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer", // Gradle module of the runtime to be started
+            Map.of( // settings
+                    "web.http.port", "9191",
+                    "edc.api.control.auth.apikey.value", API_KEY_CONTROL_AUTH,
+                    "ids.webhook.address", "http://localhost:9191"));
+
+    // EDC Provider runtime
+    @RegisterExtension
+    @Order(2)
+    static EdcExtension edc = new EdcExtension();
+
+```
+
+The extension determines the class path by running a custom Gradle task:
+
+```java
+// GradleModuleRuntimeExtension.java
+Runtime.getRuntime().exec(root + "/gradlew -q " + moduleName + ":printClasspath");
+
+// ... process classpath (see below) ...
+
+// run a thread with a custom class loader
+var classLoader = URLClassLoader.newInstance(classPathEntries,
+                                             ClassLoader.getSystemClassLoader());
+
+var mainClassName = BaseRuntime.class.getCanonicalName();
+var mainClass = classLoader.loadClass(mainClassName);
+var mainMethod = mainClass.getMethod("main", String[].class);
+
+runtimeThread = new Thread(() ->
+                           {
+                             Thread.currentThread().setContextClassLoader(classLoader);
+                             mainMethod.invoke(null, new Object[]{new String[0]});
+                           })
+```
+
+```kotlin
+     // build.gradle.kts (under allprojects)
+     tasks.register("printClasspath") {
+     doLast {
+        println("${sourceSets["main"].runtimeClasspath.asPath}");
+        }
+    }
+```
+
+The Gradle classpath is composed of JARs within the Connector project (e.g. `/path/to/EclipseDataSpaceConnector/extensions/api/control/build/libs/control-0.0.1-SNAPSHOT.jar`) and JARs from the Gradle cache (e.g. `/home/user/.gradle/caches/modules-2/files-2.1/org.eclipse.jetty/jetty-util/11.0.6/292fa5d7b2cef3483da8a7fa9dd608bfc9896564/jetty-util-11.0.6.jar`). The Extension code replaces JAR entries of Connector modules with their compiled classes and resource directories (`build/classes/java/main/` and `build/resources/main/`), and loads the connector within a separate Class Loader.
+
+Setting up the Test task to pass standard output:
+
+```kotlin
+// samples/04.0-file-transfer/integration-tests/build.gradle.kts
+tasks.getByName<Test>("test") {
+    testLogging {
+        showStandardStreams = true
+    }
+}
+```
+
+The test runs in seconds.
+
+```shell
+$ RUN_INTEGRATION_TEST=true time ./gradlew cleanTest :samples:04.0-file-transfer:integration-tests:test --tests org.eclipse.dataspaceconnector.samples.ClassLoaderWithGradleClasspathTest
```

Comment:
the `RUN_INTEGRATION_TEST` env var is not needed, because it's needed only for integration tests intended as tests that verify the integration with external systems (e.g. databases)

Reply:
Indeed, fixed